### ISSUE TITLE
accounts go into cooldown because of database lock exception

### ISSUE THREAD

**Issue by marnikvde** [RocketMap/RocketMap#1232]
**Description**
I understand that there are some reasons to put an account in cooldown mode, but there is also a "general exception" reason to put accounts in cooldown, which is probably not needed. All my accounts go in cooldown because the database seems to be locked.

Some info: I'm running this in docker on a raspberry pi, and sqlite is probably not the best approach, but it looks like a general issue.

**Possible Solution**
Be more granular on the kind of exception, and exclude this database lock of the cooldown

**Your Environment**
Version: Not sure and it's not in the log, I did a git clone on sept 7 or 8.
Environment: RaspBerryPi, running inside a docker container

logs:

```
2016-09-09 00:27:14,181 [search-worker-16][        search][    INFO] Waiting to get new account from the queue
2016-09-09 00:27:14,242 [    db-updater-0][        models][ WARNING] DB queue is > 50 (@52); try increasing --db-threads
2016-09-09 00:27:14,273 [ search-worker-9][        search][   ERROR] Exception in search_worker under account <edited> Exception message: database is locked
2016-09-09 00:27:14,274 [ search-worker-9][        search][    INFO] Waiting to get new account from the queue
2016-09-09 00:27:14,275 [ search-worker-0][        search][   ERROR] Exception in search_worker under account <edited> Exception message: database is locked
2016-09-09 00:27:14,276 [ search-worker-0][        search][    INFO] Waiting to get new account from the queue
2016-09-09 00:27:14,298 [    db-updater-0][        models][ WARNING] DB queue is > 50 (@51); try increasing --db-threads
```

```
2016-09-09 00:27:37,491 [account-recycler][        search][    INFO] Account recycler running. Checking status of 11 accounts
2016-09-09 00:27:37,495 [account-recycler][        search][    INFO] Account <edited> needs to cool off for 7177 seconds due to exception
2016-09-09 00:27:37,525 [account-recycler][        search][    INFO] Account <edited> needs to cool off for 7177 seconds due to exception
```

**Comment by lastrolo** [comment]
I have the same issue. Accounts going in to cool down because of DB locks

**Comment by Shentang** [comment]
Same here. It's kind of a blocker, unable to use the map at all.

**Comment by lastrolo** [comment]
I have an idea I already submitted on the feature hub a long time back - why not have, for people using sqlite, an option to have all database writes done in one thread? So the multiple workers do all the API interaction and heavy lifting. When they complete and have DB writes to do, those workers push the writes on to a thread safe queue. Then, there is another thread running, polling the write queue in a non blocking manner. It pulls writes off the queue in a serial manner and then commits. This should eliminate DB locking issues for sqlite

**Comment by Diblo** [comment]
I had the same problem and found a fix. Try it out at tail me if it works :+1:

**Comment by FrostTheFox** [comment]
Yes, this should probably be fixed. No, you should not keep using sqlite if DB locks are an issue.

**Comment by sebastienvercammen** [comment]
Fixed with #1267, but highly recommended to read the last comment about sqlite/MySQL.
